pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex_serviceentry_test.go

```diff
@@ -22,418 +22,525 @@ import (
 
 	"istio.io/istio/pilot/pkg/features"
 	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/slices"
+	"istio.io/istio/pkg/test"
 	"istio.io/istio/pkg/test/util/assert"
 	"istio.io/istio/pkg/workloadapi"
 )
 
 func TestAmbientIndexDuplicates(t *testing.T) {
-	s := newAmbientTestServer(t, testC, testNW)
-	s.addWorkloadEntries(t, "140.140.0.10", "name0", "sa1", map[string]string{"app": "a"})
-	s.addPods(t, "140.140.0.10", "pod0", "sa1", map[string]string{"app": "a"}, nil, true, corev1.PodRunning)
-	s.addWorkloadEntries(t, "140.140.0.10", "name1", "sa1", map[string]string{"app": "a"})
-	s.addPods(t, "140.140.0.10", "pod1", "sa1", map[string]string{"app": "a"}, nil, true, corev1.PodRunning)
-	s.assertEvent(t, s.wleXdsName("name0"), s.wleXdsName("name1"), s.podXdsName("pod0"), s.podXdsName("pod1"))
-	s.assertAddresses(t, "", "pod0", "pod1", "name0", "name1")
+	cases := []struct {
+		name         string
+		multicluster bool
+	}{
+		{
+			name: "base",
+		},
+		{
+			name:         "multicluster (enabled but unused)",
+			multicluster: true,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			if c.multicluster {
+				test.SetForTest(t, &features.EnableAmbientMultiNetwork, true)
+			}
+			s := newAmbientTestServer(t, testC, testNW)
+			s.addWorkloadEntries(t, "140.140.0.10", "name0", "sa1", map[string]string{"app": "a"})
+			s.addPods(t, "140.140.0.10", "pod0", "sa1", map[string]string{"app": "a"}, nil, true, corev1.PodRunning)
+			s.addWorkloadEntries(t, "140.140.0.10", "name1", "sa1", map[string]string{"app": "a"})
+			s.addPods(t, "140.140.0.10", "pod1", "sa1", map[string]string{"app": "a"}, nil, true, corev1.PodRunning)
+			s.assertEvent(t, s.wleXdsName("name0"), s.wleXdsName("name1"), s.podXdsName("pod0"), s.podXdsName("pod1"))
+			s.assertAddresses(t, "", "pod0", "pod1", "name0", "name1")
+		})
+	}
 }
 
+// TODO:(keithmattix): confirm ordering doesn't matter for these lookups
 func TestAmbientIndex_ServiceEntry(t *testing.T) {
-	s := newAmbientTestServer(t, testC, testNW)
+	cases := []struct {
+		name         string
+		multicluster bool
+	}{
+		{
+			name: "base",
+		},
+		{
+			name:         "multicluster (enabled but unused)",
```

Comment:
does this add valuable coverage? or is it a placeholder for a future tests which exercise multi-cluster? 

If we just want to test that turning the feature flag on doesn't break things, maybe there is a way to hack that into TestMain so we confirm that for every test.

Reply:
It's the latter; there's no TestMain for this package that I could see though. The main value of these is to ensure that when the multicluster implementation of ambientindex becomes the default (i.e. the feature flag is removed), there's no observable change in behavior